src/components/modules/readonly.ts

```diff
@@ -0,0 +1,51 @@
+import Module from '../__module';
+
+/**
+ * @module readonly
+ *
+ * Has one important method:
+ *    - {Function} toggleReadonly - Set read-only mode or toggle current state
+ *
+ * @version 1.0.0
+ *
+ * @typedef {ReadOnly} ReadOnly
+ * @property {Boolean} readOnlyEnabled - read-only state
+ */
+export default class ReadOnly extends Module {
+
+  /**
+   * Value to track read-only state
+   *
+   * @type Boolean
+   * @private
+   */
+  private readOnlyEnabled: boolean = false;
+
+  /**
+   * Set initial state
+   */
+  public async prepare(): Promise<void> {
+    this.readOnlyEnabled = this.config.readOnly;
+  }
+
+  /**
+   * Set read-only mode or toggle current state
+   *
+   * @param {Boolean} state - (optional) read-only state or toggle
+   */
+  public toggleReadOnly(state: boolean = !this.readOnlyEnabled) {
```

Comment:
better change this name, but leave `toggleReadOnly` in modules instead of `setReadOnly`.

Reply:
I would suggest to name this method just `toggle` because it is already in ReadOnly class.